src/autoRespawn/main.ts

```diff
@@ -381,30 +382,61 @@ async function handleMarkChangeEvent(event: MarkChangeEvent) {
       }
 
       const { position } = addedMark
+      const { coalition = addedMark.coalition } = command
+
+      let radius = SPAWNER_MINIMUM_DISPLACEMENT_METERS
+
+      if (command.radius) {
+        radius = command.radius
+      }
 
       // make sure spawner meets minimum displacement requirement
       const nearby = await nearbySpawners({
         position,
-        accuracy: SPAWNER_MINIMUM_DISPLACEMENT_METERS,
-        coalition: Coalition.COALITION_ALL, // TODO: take a red or blue word/string from parser here
+        accuracy: radius,
+        coalition: coalition,
       })
 
       equal(nearby.length > 0, true, 'No existing spawners found')
 
-      const [{ spawnerId }] = nearby
-
-      await spawnerDestroyed(spawnerId) // destroy
+      // if radius is given, delete each spawner inside radius
+      if (command.radius) {
+        await Promise.all(
+          nearby.map(async element => {
+            const { lat, lon, alt } = element
+            const spawnerPosition = { lat, lon, alt }
+            if (distanceFrom(position, spawnerPosition) <= radius) {
```

Comment:
it looks like the `nearbySpawners` fn already handles this check for you. this would be a duplicate https://github.com/bitbubbler/yet-another-dcs-server/blob/92c263572888650da878386fdd1ba22821c8aba5/src/db/spawners.ts#L183

Reply:
`nearbySpawners()` returns spawners in a square instead of a circle if I'm not mistaken. The distance check against the radius drops every spawner that's outside the circle.
Destroying in a circle is better and more user friendly imo, because it's easier to check via F10 in DCS (using the ruler around one point if you want to be accurate)

I think I would rather do that check in the marker handling fn instead of the db query directly, because MAYBE we run into a feature where we want the queried spawners or units in a square.

On another note:
We do that differently for units and spawners.
`nearbySpawners()` returns a distance sorted array whereas `nearbyUnits()` returns the whole query with the marker handling fn mapping it to a 2D array.

Should we change that so each works the same with the same return? Changing nearbyUnits() to return a sorted array wouldn't be too bad imo. My only concern is the additional impact of sorting the array every time although we don't need it sorted for batch destroys. But that's probably neglectable.
Changing nearbySpawners() is more work and wouldn't make sense imo, because other autoRespawn fns rely on the return being a sorted array.

Reply:
> nearbySpawners() returns spawners in a square instead of a circle if I'm not mistaken

it uses a fast square search before also doing a fast circle search on the subset found within the square. So the consumer doesn't have to do it (separation of concerns, more below)

> Destroying in a circle is better and more user friendly imo, because it's easier to check via F10 in DCS (using the ruler around one point if you want to be accurate)

destroying in a circle seems fine, we can keep it

> I think I would rather do that check in the marker handling fn instead of the db query directly, because MAYBE we run into a feature where we want the queried spawners or units in a square.

I agree that one likely wants to separate these for re-use. If there is a re-use case.

I think splitting these up is fine, but I don't think we have to split them up now. I think we just need to keep separation of concerns so splitting later isn't a nightmare if we want to (see below about separation of concerns)
We save time and effort now if we don't split them up now, it's a picking and choosing game. What to split when.

> We do that differently for units and spawners.
> nearbySpawners() returns a distance sorted array whereas nearbyUnits() returns the whole query with the marker handling fn mapping it to a 2D array.

I do like the idea of being consistent so that `spawners` and `units` have identical function naming/intent so that it's easier for someone to move around the codebase. We should plan to change this. It doesn't have to be this pr though

I think I'd lean towards both functions returning a sorted set, it seems more useful more often.

I can't see, in a "nearby" search, why one would want the default sort to be anything else but distance

**Separation of concerns**

Imo, we have 3 concerns to split here, logically, within our code.

- `find` nearby spawners using a fast db query (low accuracy, square search)
- `filtering` the low-accuracy results into high accuracy results using distance around a point (like a radius) (high accuracy, circle search)
- `consume` the now filtered spawners (either high or low accuracy)

here, in autoRespawn, we `consume` the high accuracy, already filtered, results of a fast and low accuracy db lookup. We do all of the above together to achieve a goal in a reasonable performance window.

if we take `autoRespawn` out of the conversation though, and just think about "functions that can be (re)used to build things" we still have a `find` and a `filtering` step that have nothing specific to do with `autoRespawn` (as you've pointed out)

Whether we split `find` and `filtering` into one or two functions really depends on if we need to use them separately anywhere. We can always split them pre-maturely, but there's a trade-off of time/effort vs "just shipping code" if we split everything even when not needed.

In this case, I'm concerned about keeping the `find` and `filtering`, which can be re-used, outside of the thing consuming the possibly re-usable and generic find and filtering code.